### PR TITLE
[Unit] Dark fury

### PR DESCRIPTION
**Pull request by horipse01**
A new unit: dark fury

### CODE REVIEW

**Comment by horipse01** [comment]
I have implemented my new unit "dark fury", please accept it.

File: src/battle_game/core/units/dark_fury.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+ protected:
+  void Fire();
+  void SwitchMode();
+  [[nodiscard]] const char *UnitName() const override;
+  [[nodiscard]] const char *Author() const override;
+
+  uint32_t fire_count_down_{0};
+  uint32_t switch_mode_count_down_{0};
+  bool is_berserk{false};
```
Comment (LazyJazz) [inline src/battle_game/core/units/dark_fury.h:20]: use lower_case_ ending with an underscore

**Review by LazyJazz — COMMENTED** [review]
LGTM! Fix nit

File: src/battle_game/core/units/dark_fury.h (hunk 1 above)
Comment (horipse01) [reply src/battle_game/core/units/dark_fury.h]: It has been fixed and adjusted to conform to agreement of skills

**Review by LazyJazz — DISMISSED** [review]
LGTM! Call me in wechat if you want to merge your contribution into the main branch.

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**